react/features/base/media/reducer.ts

```diff
@@ -87,6 +88,30 @@ function _audio(state: IAudioState = _AUDIO_INITIAL_MEDIA_STATE, action: AnyActi
     }
 }
 
+/**
+ * The initial common media state.
+ */
+const _COMMON_INITIAL_STATE = {};
+
+/**
+ * Reducer fot the common properties in media state.
+ *
+ * @param {ICommonState} state - Common media state.
+ * @param {Object} action - Action object.
+ * @param {string} action.type - Type of action.
+ * @returns {ICommonState}
+ */
+function _common(state: ICommonState = _COMMON_INITIAL_STATE, action: AnyAction) {
```

Comment:
Generally speaking, base/media has always represented user intent, that is, a muted a flag in here does not necesssarily mean there is a track.

The initial gUM promise is concerned with actual tracks, which fits better in base/tracks.

Reply:
Make sense! 

Now the problem is that base/tracks is just an array currently. This will include converting it into some kind of object to keep the tracks and also the promise. And this could potentially lead to a lot of changes. 

Do you insist on doing the change as part of this PR or you prefer a follow up PR with the change?